docs/tutorials/schrödinger_equation.rst

```diff
@@ -0,0 +1,276 @@
+
+    def state_probabilities(state):
+        probabilities = unp.abs(state) ** 2
+        return probabilities / unp.sum(probabilities)
+
+4. Solve the equation using existing solvers
+--------------------------------------------
+
+First, we solve the equation by using Numpy as the input and ``scipy.integrate.solve_ivp`` as a solver.
+We define the initial state, the time span for the simulation, and time point we want to simulate.
+
+.. jupyter-execute::
+
+    Z = np.array([[1,0],[0,-1]])
+    X = np.array([[0,1],[1,0]])
+    init_state = np.array([1. + 0j,0. + 0j])
+
+    t_span = [0,(N-1) * dt]
+    T = np.linspace(0,(N-1) * dt,N)
+
+We solve by using ``scipy.integrate.solve_ivp`` and plot the probabilities of each state.
+
+.. jupyter-execute::
+
+    sol = solve_ivp(rhs,t_span,init_state,args=(Z,X),method='RK45',t_eval=T)
+    probabilities = state_probabilities(sol.y)
+
+    plt.plot(sol.t, probabilities[0], label="0")
+    plt.plot(sol.t, probabilities[1], label="1")
+    plt.xlabel('T')
+    plt.ylabel('Probability')
+    plt.legend()
+    plt.show()
+
+
+Second, we solve the equation by using Jax.array as the input and ``jax.experimental.ode.odeint`` as a solver.
+
+.. jupyter-execute::
+
+    Z = jnp.array([[1,0],[0,-1]])
+    X = jnp.array([[0,1],[1,0]])
+    init_state = jnp.array([1. + 0j,0. + 0j])
+
+    t_span = [0,(N-1) * dt]
+    T = jnp.linspace(0,(N-1) * dt,N)
+
+    sol = odeint(lambda y,t: rhs(t,y,Z,X),init_state,T)
+    probabilities = state_probabilities(sol.T)
+    plt.plot(T, probabilities[0], label="0")
+    plt.plot(T, probabilities[1], label="1")
+    plt.xlabel('T')
+    plt.ylabel('Probability')
+    plt.legend()
+    plt.show()
+
+
+
+5. Register custom functions using :meth:`.Alias.register_function`
+-------------------------------------------------------------------
+
+Arraylais provides to register your own custom function using :meth:`.Alias.register_function` method.
+In this section, we introduce how to register the function, taking the Runge-Kutta method as an example.
+
+The Runge-Kutta method solves differential equations by approximating power series up to
+the foruth order terms. It allows to find solutions through numerically calculations.
+
+We define the function for the Runge-Kutta method to be used later here:
+
+.. jupyter-execute::
+
+    def runge_kutta_step(Z, X, n, state):
+        k1 = dt * rhs(n * dt, state, Z, X)
+        k2 = dt * rhs(n * dt + 0.5 * dt, state + 0.5*k1, Z, X)
+        k3 = dt * rhs(n * dt + 0.5 * dt, state + 0.5*k2, Z, X)
+        k4 = dt * rhs(n * dt + dt, state + k3, Z, X)
+        return (k1 + 2*k2 + 2*k3 + k4) / 6.
+
+JAX's strength lies in vectorized operations and parallel computations. In some cases, Python for loops
+may not effectively use JAX's optimizations. Therefore, it is recommended to avoid using the Python for 
+loops to effectively maximize JAX's strength. Using ``jax.lax.scan`` function could enable efficiently
+parallel computations.
```

Comment:
We should avoid making statements about the strengths/weaknesses of JAX. Here again I think the text can be trimmed down:

"When writing the custom solver, we write and register a version for both NumPy and JAX. The NumPy version uses standard Python loops, whereas the JAX version uses the JAX looping construct `jax.lax.scan`...."

Reply:
I updated at https://github.com/Qiskit-Extensions/arraylias/pull/25/commits/328ab5b771dca327efab87a6d5d6a871184d06de.